### PR TITLE
Check if application exists before creating a new one in CI workflow

### PR DESCRIPTION
**Pull request by a-noyass**

### CODE REVIEW

**Review by mohamedshabanofficial — APPROVED** [review]

**Comment by a-noyass** [comment]
@AmirGeorge After updating BF CLI the workflow failed when trying to create an application with the same as an existing one instead of a soft failure. So now we check if it exists before creating a new application.

File: .github/workflows/luis_ci.yaml [hunk 1]
```diff
@@ -86,14 +86,23 @@ jobs:
     - name: Ludown to LUIS model
       run: bf luis:convert -i $LU_FILE -o ./model.json --name 'LUIS CI pipeline - ${{ github.run_id }}' --versionid $LUISAppVersion
 
-    # When doing a merge to master, use the master LUIS app - create if necessary (soft fails if exists)
+    # When doing a merge to master, use the master LUIS app
     - name: Get master LUIS application ID
       run: |
-        bf luis:application:create --name $LUIS_MASTER_APP_NAME --subscriptionKey $LUISAuthoringKey  --endpoint $LUISAuthoringEndpoint --versionId=0.1
```
Comment (AmirGeorge) [inline .github/workflows/luis_ci.yaml:92]: How does this change guarantee that the app already exists at this stage?

File: .github/workflows/luis_ci.yaml [hunk 2]
```diff
@@ -86,14 +86,23 @@ jobs:
...
         bf luis:application:list --subscriptionKey $LUISAuthoringKey  --endpoint $LUISAuthoringEndpoint | \
         jq -c '.[] | select(.name | . and contains('\"$LUIS_MASTER_APP_NAME\"')) | .id' | \
         xargs -I {} echo "LUISAppId={}" >> $GITHUB_ENV
 
+    # create application if necessary
+    - name: Create application if doesn't exist
+      run: |
+        if [ ${#LUISAppId} -ne 36 ]; then
```
Comment (AmirGeorge) [inline .github/workflows/luis_ci.yaml:99]: who is 36? :D

**Review by AmirGeorge — APPROVED** [review]

File: .github/workflows/luis_ci.yaml (hunk 2 above)
Comment (a-noyass) [reply .github/workflows/luis_ci.yaml:99]: length of the GUID

File: .github/workflows/luis_ci.yaml (hunk 1 above)
Comment (a-noyass) [reply .github/workflows/luis_ci.yaml:92]: @AmirGeorge  creating the application is moved to the next step and is only executed if the application is not found in this step which tries to find the application by its name

**Pull request merged**